recipes/llvm-core/all/conanfile.py

```diff
@@ -0,0 +1,283 @@
+        'rtti': [True, False],
+        'threads': [True, False],
+        'lto': ['On', 'Off', 'Full', 'Thin'],
+        'static_stdlib': [True, False],
+        'unwind_tables': [True, False],
+        'expensive_checks': [True, False],
+        'use_perf': [True, False],
+        'with_ffi': [True, False],
+        'with_zlib': [True, False],
+        'with_xml2': [True, False]
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'components': 'all',
+        'targets': 'all',
+        'exceptions': True,
+        'rtti': True,
+        'threads': True,
+        'lto': 'Off',
+        'static_stdlib': False,
+        'unwind_tables': True,
+        'expensive_checks': False,
+        'use_perf': False,
+        'with_ffi': False,
+        'with_zlib': True,
+        'with_xml2': True
+    }
+
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+    generators = ['cmake', 'cmake_find_package']
+    no_copy_source = True
+
+    _source_subfolder = 'source'
+
+    def _supports_compiler(self):
+        compiler = self.settings.compiler.value
+        version = tools.Version(self.settings.compiler.version)
+        major_rev, minor_rev = int(version.major), int(version.minor)
+
+        unsupported_combinations = [
+            [compiler == 'gcc', major_rev == 5, minor_rev < 1],
+            [compiler == 'gcc', major_rev < 5],
+            [compiler == 'clang', major_rev < 4],
+            [compiler == 'apple-clang', major_rev < 9],
+            [compiler == 'Visual Studio', major_rev < 15]
+        ]
+        if any(all(combination) for combination in unsupported_combinations):
+            message = 'unsupported compiler: "{}", version "{}"'
+            raise ConanInvalidConfiguration(message.format(compiler, version))
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get('patches', {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _patch_build(self):
+        if Path('FindIconv.cmake').exists():
+            tools.replace_in_file('FindIconv.cmake', 'iconv charset', 'iconv')
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions['BUILD_SHARED_LIBS'] = False
+        cmake.definitions['CMAKE_SKIP_RPATH'] = True
+
+        cmake.definitions['LLVM_TARGETS_TO_BUILD'] = self.options.targets
+        cmake.definitions['LLVM_TARGET_ARCH'] = 'host'
+        cmake.definitions['LLVM_BUILD_LLVM_DYLIB'] = self.options.shared
+        cmake.definitions['LLVM_DYLIB_COMPONENTS'] = self.options.components
+        cmake.definitions['LLVM_ENABLE_PIC'] = \
+            self.options.get_safe('fPIC', default=False)
+
+        cmake.definitions['LLVM_ABI_BREAKING_CHECKS'] = 'WITH_ASSERTS'
+        cmake.definitions['LLVM_ENABLE_WARNINGS'] = True
+        cmake.definitions['LLVM_ENABLE_PEDANTIC'] = True
+        cmake.definitions['LLVM_ENABLE_WERROR'] = False
+
+        cmake.definitions['LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN'] = True
```

Comment:
While we wait for the **CI** confirmation we have all green, this is the only thing left in my opinion. This flag is to avoid failure on the **Visual Studio 2019** builds with the version we currently have (**16.4**) because of [this issue](https://github.com/conan-io/conan-center-index/pull/1450#discussion_r415158802).

I'd like to disable it before merging to master as I'm not sure what "*Visual Studio version 16.4 and lower is known to miscompile part of LLVM*" implies, so better to wait until we have an updated version in the **CI**. A hard error won't pass unnoticed to someone trying to build a package using an old version.